payments-core/src/main/java/com/stripe/android/payments/core/authentication/WebIntentAuthenticator.kt

```diff
@@ -96,31 +96,10 @@ internal class WebIntentAuthenticator @Inject constructor(
                 authUrl = nextActionData.webViewUrl.toString()
                 returnUrl = nextActionData.returnUrl
             }
-            is StripeIntent.NextActionData.DisplayOxxoDetails -> {
+            is StripeIntent.NextActionData.DisplayVoucherDetails -> {
                 // nextActionData.hostedVoucherUrl will never be null as AuthenticatorRegistry won't direct it here
                 authUrl = nextActionData.hostedVoucherUrl.takeIf { it!!.isNotEmpty() }
-                    ?: throw IllegalArgumentException("null hostedVoucherUrl for DisplayOxxoDetails")
-                returnUrl = null
-                shouldCancelIntentOnUserNavigation = false
-            }
-            is StripeIntent.NextActionData.DisplayBoletoDetails -> {
-                // nextActionData.hostedVoucherUrl will never be null as AuthenticatorRegistry won't direct it here
-                authUrl = nextActionData.hostedVoucherUrl.takeIf { it!!.isNotEmpty() }
-                    ?: throw IllegalArgumentException("null hostedVoucherUrl for DisplayBoletoDetails")
-                returnUrl = null
-                shouldCancelIntentOnUserNavigation = false
-            }
-            is StripeIntent.NextActionData.DisplayKonbiniDetails -> {
-                // nextActionData.hostedVoucherUrl will never be null as AuthenticatorRegistry won't direct it here
-                authUrl = nextActionData.hostedVoucherUrl.takeIf { it!!.isNotEmpty() }
-                    ?: throw IllegalArgumentException("null hostedVoucherUrl for DisplayKonbiniDetails")
-                returnUrl = null
-                shouldCancelIntentOnUserNavigation = false
-            }
-            is StripeIntent.NextActionData.DisplayMultibancoDetails -> {
-                // nextActionData.hostedVoucherUrl will never be null as AuthenticatorRegistry won't direct it here
-                authUrl = nextActionData.hostedVoucherUrl.takeIf { it!!.isNotEmpty() }
-                    ?: throw IllegalArgumentException("null hostedVoucherUrl for DisplayMultibancoDetails")
+                    ?: throw IllegalArgumentException("null hostedVoucherUrl for DisplayVoucherDetails")
```

Comment:
We lose type information here as well, but again I think the tradeoff is worth it here given how few voucher LPMs there are. Lmk if you have any better ideas